eng/common/pipelines/templates/validate-filename.yml

```diff
@@ -0,0 +1,15 @@
+parameters:
+  WorkingDirectory: '$(System.DefaultWorkingDirectory)'
+steps:
+  - pwsh: |
+      $differByCaseFiles = git ls-files | Group-Object | Where-Object { $_.Count -gt 1 } 
+
+      if ($differByCaseFiles) 
+      {
+        Write-Error "Do NOT name the files which only differ in case. Please check and rename the following files:"
+        $differByCaseFiles | ForEach-Object { Write-Error "Duplicated Files: [$($_.Group)]"}
```

Comment:
For devops the first Write-Error will stop the rest of the from outputting. Instead we should just Write-Host and then do one Write-Error at the end. 

Also you should prefer a foreach(...) { } loop over using the ForEach-Object cmdlet.